### PR TITLE
Add GetClosest point solution in Circle class.

### PR DESCRIPTION
**Pull request by thiamchi**
https://github.com/20chan/cirno/issues
I had give the idea and method to get the closest point. Should work. =D

### CODE REVIEW

**Comment by 20chan** [comment]
@thiamchi Thanks for this started! Can you add unit tests for this method?

File: cirno/Geometry/Circle.cs [hunk 1]
```diff
@@ -59,6 +59,52 @@ public Vector GetClosestPoint(Vector point)
             return d1 < d2 ? v1 : v2;
         }
 
+        public float Distance(Vector point)
```
Comment (20chan) [inline cirno/Geometry/Circle.cs:62]: Your `Distance` method is exactly same as `GetClosestPoint` method.
Can you remove duplicates?

```csharp
public float Distance(Vector point) {
    return Vector.Distance(point, GetClosestPoint(point));
}
```

There is no `Vector.Distance`, so you should implement it.

Comment (thiamchi) [reply cirno/Geometry/Circle.cs:62]: OK. I will remove later. I should add distance function. Having dinner. =D

**Comment by thiamchi** [comment]
OK. Mistake was made but fixed. Added Unit Test.

**Comment by thiamchi** [comment]
For Line to Circle, try to get a perpendicular line that pass through the center of circle and solve with quadratic equation. Note: Gradient of given line and the perpendicular line have a product equal to -1.
For Circle to circle, try to define center points in comments like what I did in GetClosestPoint function. Then, again find solutions from quadratic equation again. (Probably can have a class for solving equation?)
Happy Hacktober!

Reference : https://www.analyzemath.com/CircleEq/circle_intersection.html

File: cirno/Vector.cs [hunk 2]
```diff
@@ -32,6 +32,11 @@ public struct Vector {
                 && Math.Abs(Y - other.Y) < tolerance;
         }
 
+        public float Distance(Vector point)
+        {
+            return (float) Math.Sqrt(Math.Pow(X - point.X, 2) + Math.Pow(Y - point.Y, 2));
```
Comment (20chan) [inline cirno/Vector.cs]: Just one more formatting issue, 
```csharp
return (float) Math.Sqrt
```
Can you remove whitespace after type casts?

Comment (thiamchi) [reply cirno/Vector.cs]: Done

Comment (20chan) [reply cirno/Vector.cs]: Thanks for fast response, btw, before you comment I already merged lol

**Pull request merged**